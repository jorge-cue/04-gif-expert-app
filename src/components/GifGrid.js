import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';
// import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>
            <div className="card-grid">
            
            { loading && <p>Loading...</p> }
            {
                images.map( gif => ( 
                    <GifGridItem 
                        key= {gif.id} 
                        { ...gif } 
                    /> ))
            }
            </div>
        </>
    )
}
