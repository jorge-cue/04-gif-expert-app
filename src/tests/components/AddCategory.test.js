import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from "../../components/AddCategory";


describe('AddCategory test suite', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    });

    test('should render successfully', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should change text box', () => {
        const value = 'Hello World!';
        wrapper.find('input').simulate('change', { target: { value } });
        expect( wrapper.find('input').prop('value') ).toBe( value );
    });

    test('should not submit if input value is shorter than 2 characters', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect( setCategories ).not.toHaveBeenCalled()
    });

    test('should submit if input value is at least 2 characters', () => {
        const value = 'Hello world!';
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
        expect ( wrapper.find('input').prop('value') ).toBe( '' );
    });
})