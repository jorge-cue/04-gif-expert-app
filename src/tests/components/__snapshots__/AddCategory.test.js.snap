// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory test suite should render successfully 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
